pkgs/applications/misc/ethereum/cpp/default.nix

```diff
@@ -0,0 +1,134 @@
+{ stdenv
+, fetchgit
+, cmake
+, boost
+, gmp
+, jsoncpp
+, leveldb
+, cryptopp
+, cudatoolkit
+, nvidia_x11
+, libcpuid
+, miniupnpc
+, libjson_rpc_cpp
+, curl
+, libmicrohttpd
+, mesa
+, qtwebengine ? null
+, qtbase ? null
+, qtdeclarative ? null
+}:
+let
+  withGUI = (qtwebengine != null) && (qtbase != null) && (qtdeclarative != null);
```

Comment:
Could you make this a flag that defaults to false? If someone adds all of these in top level "pkgs" callPackage will provide them and it will enable GUI without being obvious. Instead I think this should be an assertion like:

```
assert withGUII -> (qtwebengine != null) && (qtbase != null) && (qtdeclarative != null);
```